File: API/Entities/Sale.cs
using System;

namespace API.Entities
{
    public class Sale
    {
        public int Id { get; set; }
        public Product Product { get; set; }
        public SalesPerson SalesPerson { get; set; }
        public Customer Customer { get; set; }
        public DateTime SaleDate { get; set; }

    }
}

File: client/src/app/_models/sale.ts
import { Customer } from "./customer";
import { Product } from "./product";
import { SalesPerson } from "./salesperson";

export interface Sale {
    productId: number;
    product?: Product;
    salesPersonId: number;
    salesPerson?: SalesPerson;
    customerId: number;
    customer?: Customer;
    saleDate?: string;
}